include/zephyr/drivers/interrupt_controller/riscv_plic.h

```diff
@@ -12,27 +12,77 @@
 #ifndef ZEPHYR_INCLUDE_DRIVERS_RISCV_PLIC_H_
 #define ZEPHYR_INCLUDE_DRIVERS_RISCV_PLIC_H_
 
+#include <zephyr/device.h>
+
+/**
+ * @brief Install the ISR into the table
+ *
+ * @param irq Interrupt ID
+ * @param routine Pointer to the ISR handler
+ * @param param Pointer to the argument to be passed into the ISR handler
+ */
+void z_riscv_plic_isr_install(unsigned int irq, unsigned int priority,
+			      void (*routine)(const void *), const void *param);
+
+/**
+ * @brief Install the ISR of an interrupt controller into the table
+ * 
+ * See @ref z_riscv_plic_isr_install for more info
+ *
+ * @param intc Interrupt controller of the irq, the first instance will be used if this is NULL
+ */
+void z_riscv_plic_isr_intc_install(const struct device *intc, unsigned int irq,
+				   unsigned int priority, void (*routine)(const void *),
+				   const void *param);
+
 /**
  * @brief Enable interrupt
  *
  * @param irq interrupt ID
  */
-void riscv_plic_irq_enable(uint32_t irq);
+void riscv_plic_irq_enable(uint32_t irq, const struct device *dev);
+
+/**
+ * @brief Enable interrupt of an interrupt controller
+ *
+ * See @ref riscv_plic_irq_enable for more info
+ *
+ * @param intc Interrupt controller of the irq, the first instance will be used if this is NULL
+ */
+void riscv_plic_irq_intc_enable(const struct device *intc, uint32_t irq);
 
 /**
  * @brief Disable interrupt
  *
  * @param irq interrupt ID
  */
-void riscv_plic_irq_disable(uint32_t irq);
+void riscv_plic_irq_disable(uint32_t irq, const struct device *dev);
+
+/**
+ * @brief Disable interrupt of an interrupt controller
+ *
+ * See @ref riscv_plic_irq_disable for more info
+ *
+ * @param intc Interrupt controller of the irq, the first instance will be used if this is NULL
+ */
+void riscv_plic_irq_intc_disable(const struct device *intc, uint32_t irq);
 
 /**
  * @brief Check if an interrupt is enabled
  *
  * @param irq interrupt ID
  * @return Returns true if interrupt is enabled, false otherwise
  */
-int riscv_plic_irq_is_enabled(uint32_t irq);
+int riscv_plic_irq_is_enabled(uint32_t irq, const struct device *dev);
```

Comment:
the signature for this should probably not be changed.

Reply:
as the `irq` doesnt carry the interrupt controller info, an additional `dev` arg is required in a multi-instance configuration